cmd/minishift/cmd/start.go

```diff
@@ -127,6 +128,12 @@ func runStart(cmd *cobra.Command, args []string) {
 	setOcProxy()
 	setShellProxy()
 
+	if viper.IsSet("http-proxy") {
+		proxyUrl = viper.GetString(httpProxy)
```

Comment:
You seem to misunderstand me. I am not saying you should use _RegistrationParameters_ to for the Docker and OpenShift things. But you should use them to pass these parameters into the registration algorithm. _ReigsitrationParamters_ is using the parameter object pattern. It encapsulates the parameters needed as part of the registration. This is independent/orthogonal  to what you do with the OpenShift and Docker parameters.